esmvaltool/utils/cmorize_obs.py

```diff
@@ -0,0 +1,273 @@
+    logger.info("input_dir  = %s", config_user["rootpath"]["RAWOBS"][0])
+    logger.info("output_dir = %s", config_user["output_dir"])
+    logger.info(70 * "-")
+
+    # call the reformat function
+    if args.obs_list_cmorize:
+        obs_list = args.obs_list_cmorize
+    else:
+        obs_list = []
+    _cmor_reformat(config_user, obs_list)
+
+    # End time timing
+    timestamp2 = datetime.datetime.utcnow()
+    logger.info(
+        "Ending the CMORization Tool at time: %s UTC",
+        timestamp2.strftime(timestamp_format))
+    logger.info(
+        "Time for running the CMORization scripts was: %s",
+        timestamp2 - timestamp1)
+
+
+def _cmor_reformat(config, obs_list):
+    """Run the cmorization routine."""
+    logger.info("Running the CMORization scripts.")
+
+    # master directory
+    raw_obs = config["rootpath"]["RAWOBS"][0]
+
+    # set the reformat scripts dir
+    reformat_scripts = os.path.join(os.path.dirname(__file__),
+                                    '../cmor/cmorizers/obs')
+    run_dir = os.path.join(config['output_dir'], 'run')
+    # datsets dictionary of Tier keys
+    datasets = _assemble_datasets(raw_obs, obs_list)
+    logger.info("Processing datasets %s", datasets)
+
+    # loop through tier/datasets to be cmorized
+    for tier, _ in datasets.items():
+        for dataset in datasets[tier]:
+            reformat_script_root = os.path.join(reformat_scripts,
+                                                'cmorize_obs_' + dataset)
+            # in-data dir; build out-dir tree
+            in_data_dir = os.path.join(raw_obs, tier, dataset)
+            out_data_dir = os.path.join(config['output_dir'], tier, dataset)
+            if not os.path.isdir(out_data_dir):
+                os.makedirs(out_data_dir)
+
+            # all operations are done in the working dir now
+            os.chdir(out_data_dir)
+
+            # figure out what language the script is in
+            if os.path.isfile(reformat_script_root + '.ncl'):
+                reformat_script = reformat_script_root + '.ncl'
+                logger.info("CMORizing dataset %s using NCL script %s",
+                            dataset, reformat_script)
+                # copy over the reformat script
+                shutil.copy2(reformat_script, out_data_dir)
+                # call the ncl script
+                _run_ncl_script(in_data_dir,
+                                out_data_dir,
+                                run_dir,
+                                dataset,
+                                reformat_script,
+                                config['log_level'])
+            elif os.path.isfile(reformat_script_root + '.py'):
+                py_reformat_script = reformat_script_root + '.py'
+                logger.info("CMORizing dataset %s using Python script %s",
+                            dataset, py_reformat_script)
+                # copy over the reformat script
+                shutil.copy2(py_reformat_script,
```

Comment:
Why are you copying and then importing? This may lead non-reproducible results if people start editing those files and re-running. Best to just import from the installed esmvaltool.

Reply:
same as with the ncl script, copying is fine coz it gets overwritten with the installed script everytime one runs :)

Reply:
le fixed